crates/turborepo-lib/src/shim.rs

```diff
@@ -403,20 +395,20 @@ impl LocalTurboState {
         {
             // Needs borrow because of the loop.
             #[allow(clippy::needless_borrow)]
-            let bin_path = root.join(&platform_package_executable_path);
+            let bin_path = root.join_components(&platform_package_executable_path_components);
             match fs_canonicalize(&bin_path) {
                 Ok(bin_path) => {
-                    let resolved_package_json_path = root.join(platform_package_json_path);
-                    let platform_package_json_string =
-                        fs::read_to_string(resolved_package_json_path).ok()?;
-                    let platform_package_json: PackageJson =
-                        serde_json::from_str(&platform_package_json_string).ok()?;
```

Comment:
As a side effect of using a unified `PackageJson` and `PackageJson::load` we remove the simplistic `serde_json` approach. Almost certainly a tiny bit slower but this gets called a maximum of one time and consistent parsing of `package.json` is better.